internal/authentication/ldap_client_factory.go

```diff
@@ -117,162 +121,393 @@ type PooledLDAPClientFactory struct {
 	opts   []ldap.DialOpt
 	dialer LDAPClientDialer
 
-	pool chan *LDAPClientPooled
-	mu   sync.Mutex
-
-	sleep time.Duration
+	// Pool management
+	pool           chan *LDAPClientPooled // Channel for available clients
+	activeCount    atomic.Int32           // Atomic counter for active connections
+	minPoolSize    int32                  // Minimum number of pool connections to maintain (soft target)
+	maxPoolSize    int32                  // Maximum number of pool connections (hard target)
+	clientLifetime time.Duration          // Maximum lifetime for a pooled client (soft target)
+
+	// Synchronization
+	wakeup  chan struct{}      // Channel for client requests
+	closing atomic.Bool        // Atomic flag indicating if the pool is closing
+	closed  chan error         // Signals when poolManager has completed cleanup
+	ctx     context.Context    // Context for cancellation
+	cancel  context.CancelFunc // Function to cancel the context
+}
 
-	closing bool
+func (f *PooledLDAPClientFactory) isClosing() bool {
+	return f.closing.Load()
 }
 
+// initializes the pool and starts the pool manager goroutine
+// must be called only once
 func (f *PooledLDAPClientFactory) Initialize() (err error) {
-	f.mu.Lock()
-
-	defer f.mu.Unlock()
-
-	if f.pool != nil {
-		return nil
+	if f.pool != nil { // Already initialized
+		return fmt.Errorf("LDAP pool is already initialized")
 	}
 
-	f.pool = make(chan *LDAPClientPooled, f.config.Pooling.Count)
-
-	var (
-		errs   []error
-		client *LDAPClientPooled
-	)
+	minRecommendedTimeout := f.config.Timeout * time.Duration(f.config.Pooling.Retries+2) // +2 to account for initial attempt and backoff
+	if f.config.Pooling.Timeout <= minRecommendedTimeout {
+		logging.Logger().Warnf("LDAP pooling timeout (%v) may be insufficient for retry scenarios (recommended: >%v)",
```

Comment:
Please remove call to `logging.Logger()` and initialize a fielded logger like in #10383, include the braces in another calll to WithFields, and avoid using Warnf and use Warn instead.